### PR TITLE
feat: market liquidities account

### PR DESCRIPTION
**Pull request by pawel-betdex**
Second account needed to deliver improved matching. This account holds summary of the unmatched liquidity available on the market.

### CODE REVIEW

**Review by obi-ewan — DISMISSED** [review]

File: programs/monaco_protocol/src/state/market_liquidities.rs [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+            .binary_search_by(Self::sorter_for(outcome, price))
+        {
+            Ok(index) => self.liquidities_for[index],
+            Err(_) => MarketOutcomePriceLiquidity {
+                outcome,
+                price,
+                liquidity: 0,
+            },
+        }
+    }
```
Comment (eoin-betdex) [inline programs/monaco_protocol/src/state/market_liquidities.rs:48]:
```suggestion
    pub fn get_liquidity_against(&self, outcome: u16, price: f64) -> MarketOutcomePriceLiquidity {
        match self
            .liquidities_against
            .binary_search_by(Self::sorter_against(outcome, price))
        {
            Ok(index) => self.liquidities_against[index],
            Err(_) => MarketOutcomePriceLiquidity {
                outcome,
                price,
                liquidity: 0,
            },
        }
    }
```
That's what you get for skipping unit tests!

File: programs/monaco_protocol/src/state/market_liquidities.rs [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+    pub market: Pubkey,
+    pub liquidities_for: Vec<MarketOutcomePriceLiquidity>,
+    pub liquidities_against: Vec<MarketOutcomePriceLiquidity>,
+}
+
+impl MarketLiquidities {
+    pub const SIZE: usize = DISCRIMINATOR_SIZE
+        + PUB_KEY_SIZE // market
+        + vec_size(18_usize, 0_usize) // liquidities_for
+        + vec_size(18_usize, 0_usize); // liquidities_against
```
Comment (eoin-betdex) [inline programs/monaco_protocol/src/state/market_liquidities.rs:20]: NP `18` is a bit of a magic number here. Usually I think we would implement size_of for the struct in question, and use that here but it's fine.

**Review by eoin-betdex — CHANGES_REQUESTED** [review]
Looks good I think. Just maybe a copy and paste error left in `get_liquidity_against`

File: programs/monaco_protocol/src/state/market_liquidities.rs (hunk 1 above)
Comment (pawel-betdex) [reply programs/monaco_protocol/src/state/market_liquidities.rs:48]: yes

File: programs/monaco_protocol/src/state/market_liquidities.rs (hunk 2 above)
Comment (pawel-betdex) [reply programs/monaco_protocol/src/state/market_liquidities.rs:20]: Ok. Let's do it now.

**Review by eoin-betdex — APPROVED** [review]

**Review by obi-ewan — APPROVED** [review]

**Pull request merged**